docs/rfcs/confmap-provider-configuration.md

```diff
@@ -0,0 +1,246 @@
+# Configuration of confmap Providers
+
+## Motivation
+
+The `confmap.Provider` interface is used by the Collector to retrieve map
+objects representing the Collector's configuration or a subset thereof. Sources
+config may include locally-available information such as files on disk or
+environment variables, or may be remotely accessed over the network. In the
+process of obtaining configuration from a source, the user may wish to modify
+the behavior of how the source is obtained.
+
+For example, consider the case where the Collector obtains configuration over
+HTTP from an HTTP endpoint. A user may want to:
+
+1. Poll the HTTP endpoint for configuration at a configurable interval and
+   reload the Collector service if the configuration changes.
+2. Authenticate the request to get configuration by including a header in the
+   request.
+
+This would produce a set of options like the following:
+
+- `poll-interval`: Sets an interval for the Provider to check the HTTP endpoint
+  for changes. If the config has changed, the service will be reloaded. Defaults
+  to `0`, which disables polling.
+- `headers`: Specifies a semicolon-delimited set of headers that will be passed
+  to the config backend. Header keys and values should be separated by `=`.
+
+Configuration for Providers cannot be stored in the same place as the rest of
+the Collector's configuration since the Providers are the ones tasked with
+getting that configuration. Therefore, they need a separate configuration
+mechanism.
+
+## Current state
+
+No upstream Providers currently offer any configuration options. The exported
+interfaces are still able to change before the `confmap` module is declared
+stable, but avoiding breaking changes in the API would be preferable.
+
+## Desired state
+
+We would like a mechanism to configure Providers that meets the following
+requirements:
+
+1. Broadly applicable to all current upstream providers.
+2. Can be used as a suggestion for configuring custom providers.
+3. Is consistent between Providers.
+4. Is configurable per config URI.
```

Comment:
I feel like a requirement should be that any valid URI today is expressible after this change (possibly with some escaping mechanism)

Reply:
Agreed. It looks like something like `##` wouldn't be a valid normal URI fragment [according to RFC 3986](https://www.rfc-editor.org/rfc/rfc3986#appendix-A), so that may be a good default for at least the URI option. I think the other options won't need to mess with the URI.